src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
This code is a bit hard to read. I'd suggest to refactor it this way: 

```suggestion
  const women = people.filter((p) => p.sex === 'f');
  
  const filteredWomen = withChildren 
    ? women.filter(woman => people.find(person => person.mother === person.name))
    : women;
```

Should work the same way and looks a little cleaner.
1. To avoid repeatable code we move filter logic to separate variable.
2. We can ommit parentheses if your function takes a single argument and doesn't use braces.
3. For better readability you should avoid naming variables and arguments with one letter.